framework/Optimizers/Optimizer.py

```diff
@@ -400,11 +412,19 @@ def initialize(self,externalSeeding=None,solutionExport=None):
     self.counter['mdlEval'] = 0
     self.counter['varsUpdate'] = [0]*len(self.optTraj)
 
+    for entry in self.assemblerDict.get('Preconditioner',[]):
+      cls,typ,name,model = entry
+      if cls != 'Models' or typ != 'ExternalModel':
+        self.raiseAnError(IOError,'Currently only "ExternalModel" models can be used as preconditioners! Got "{}.{}" for "{}".'.format(cls,typ,name))
+      self.preconditioners[name] = model
+
     self.mdlEvalHist = self.assemblerDict['TargetEvaluation'][0][3]
     self.objSearchingROM = SupervisedLearning.returnInstance('SciKitLearn', self, **{'SKLtype':'neighbors|KNeighborsRegressor', 'Features':','.join(list(self.fullOptVars)), 'Target':self.objVar, 'n_neighbors':1,'weights':'distance'})
     self.solutionExport = solutionExport
+    if self.solutionExport is None:
+      self.raiseAnError(IOError,'The results of optimization cannot be obtained with a SolutionExport in the Step!')
```

Comment:
I am not sure that this error makes sense (as in it seems to say you should not have a SolutionExport, but is checking that there is no solutionExport.)  Maybe I am misunderstanding.